src/ui/win32/bindings/MemoryViewerControlBinding.cpp

```diff
@@ -264,6 +268,15 @@ void MemoryViewerControlBinding::OnClick(POINT point)
     Invalidate();
 }
 
+void MemoryViewerControlBinding::OnShiftClick(POINT point)
+{
+    OnClick(point);
+    const auto& pEmulatorContext = ra::services::ServiceLocator::Get<ra::data::context::EmulatorContext>();
+    ra::ByteAddress nAddress =
+        pEmulatorContext.ReadMemory(m_pViewModel.GetAddress(), m_pViewModel.GetSize()) & 0xFFFFFF;
+    m_pViewModel.OnShiftClick(nAddress);
```

Comment:
You're not shift+clicking `nAddress`, you're just setting the address after converting it. The conversion should be here, then just call `SetAddress`.

Although, for unit testing purposes all of this code should be in the view model. Suggest making `MemoryViewerViewModel::OnShiftClick` accept an x and y, have that call `MemoryViewerViewModel::OnClick`, do the address calculations and call `MemoryViewerViewModel::SetAddress`. Then, please, write unit tests for `MemoryViewerViewModel::OnShiftClick`.